### PR TITLE
feat: gate with daylylimit and freeze

### PR DESCRIPTION
**Pull request by sokel**

### CODE REVIEW

File: blockchain/source/contracts/SimpleGatekeeperWithLimit.sol [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+
+        if (paid[txHash].commitTS == 0) {
+            // check daylimit
+            require(underLimit(msg.sender, _value));
+            paid[txHash].commitTS = block.timestamp;
+            paid[txHash].keeper = msg.sender;
+            emit CommitTx(_to, _txNumber, _value, block.timestamp);
+        } else {
+            require(paid[txHash].keeper == msg.sender);
+            require(paid[txHash].commitTS + 1 days >= block.timestamp);
```
Comment (sshaman1101) [inline blockchain/source/contracts/SimpleGatekeeperWithLimit.sol]: seems you must check that commitTS + 1 day is not before today.

File: blockchain/source/contracts/SimpleGatekeeperWithLimit.sol [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+        return false;
+    }
+
+    function today() private view returns (uint256) {
+        // solium-disable-next-line security/no-block-members
+        return block.timestamp / 1 days;
+    }
+
+    function kill() public onlyOwner {
+        uint balance = token.balanceOf(this);
```
Comment (sshaman1101) [inline blockchain/source/contracts/SimpleGatekeeperWithLimit.sol]: uint256

File: blockchain/source/contracts/SimpleGatekeeperWithLimitLive.sol [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+    }
+
+    function UnFreezeKeeper(address _keeper) public onlyOwner {
+        require(keepers[_keeper].dayLimit > 0);
+        keepers[_keeper].frozen = false;
+        emit KeeperUnfrozen(_keeper);
+    }
+
+    function Payin(uint256 _value) public {
+        token.transferFrom(msg.sender, this, _value);
```
Comment (sshaman1101) [inline blockchain/source/contracts/SimpleGatekeeperWithLimitLive.sol]: require?

File: blockchain/source/contracts/SimpleGatekeeperWithLimitLive.sol [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+        if (paid[txHash].commitTS == 0) {
+            // check daylimit
+            require(underLimit(msg.sender, _value));
+            paid[txHash].commitTS = block.timestamp;
+            paid[txHash].keeper = msg.sender;
+            emit CommitTx(_to, _txNumber, _value, block.timestamp);
+        } else {
+            require(paid[txHash].keeper == msg.sender);
+            require(paid[txHash].commitTS + 1 days >= block.timestamp);
+            token.transfer(_to, _value);
```
Comment (sshaman1101) [inline blockchain/source/contracts/SimpleGatekeeperWithLimitLive.sol]: require?

File: blockchain/source/contracts/SimpleGatekeeperWithLimitLive.sol [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+    }
+
+    function today() private view returns (uint256) {
+        // solium-disable-next-line security/no-block-members
+        return block.timestamp / 1 days;
+    }
+
+    function kill() public onlyOwner {
+        uint balance = token.balanceOf(this);
+        token.transfer(owner, balance);
```
Comment (sshaman1101) [inline blockchain/source/contracts/SimpleGatekeeperWithLimitLive.sol]: require?

File: blockchain/source/test/simpleGatekeeperWithLimit.js [hunk 6]
```diff
@@ -0,0 +1,191 @@
...
+        let token;
+        let gatekeeper;
+
+        let owner = accounts[0];
+        let user = accounts[1];
+
+        let tx;
+
+        let startGatekeeperBalance;
+        let startUserBalance;
```
Comment (sshaman1101) [inline blockchain/source/test/simpleGatekeeperWithLimit.js]: it's initial balance, not a "start" one.

**Comment by sokel** [comment]
Change some names.
PTAL again. thx.

**Pull request merged**